spark-gremlin/src/main/java/org/apache/tinkerpop/gremlin/spark/process/computer/SparkGraphComputer.java

```diff
@@ -130,12 +142,13 @@ public GraphComputer config(final String key, final Object value) {
                 try {
                     graphRDD = hadoopConfiguration.getClass(Constants.GREMLIN_SPARK_GRAPH_INPUT_RDD, InputFormatRDD.class, InputRDD.class)
                             .newInstance()
-                            .readGraphRDD(this.sparkConfiguration, sparkContext)
-                            .setName("graphRDD")
+                            .readGraphRDD(apacheConfiguration, sparkContext)
+                            .setName(sparkConfiguration.get(Constants.GREMLIN_HADOOP_OUTPUT_LOCATION, "graphRDD"))
                             .cache();
```

Comment:
I believe this "cache" is basically saving the initial state of the graph after having been read from the source. This will be helpful if there are two separate DAGs being built from the graphRDD base, but we need to be sure to unpersist this in all cases.